src/guides/v2.3/frontend-dev-guide/templates/template-email-migration.md

```diff
@@ -0,0 +1,230 @@
+    /**
+    * Calculates the lifetime spend of all customers
+    */
+    class LifetimeSpendDirective implements ProcessorInterface
+    {
+        /**
+        * @var PriceCurrencyInterface
+        */
+        private $priceCurrency;
+        /**
+        * @param PriceCurrencyInterface $priceCurrency
+        */
+        public function __construct(PriceCurrencyInterface $priceCurrency)
+        {
+            $this->priceCurrency = $priceCurrency;
+        }
+        /**
+        * @inheritDoc
+        */
+        public function getName(): string
+        {
+            return 'lifetime_spend';
+        }
+        /**
+        * @inheritDoc
+        */
+        public function process($value, array $parameters, ?string $html): string
+        {
+            $shouldBold = !empty($parameters['should_bold']);
+            $amount = $this->priceCurrency->getCurrencySymbol() . $this->calculateLifetimeSpend();
+            return ($shouldBold ? '<strong>' . $amount . '</strong>' : $amount);
+        }
+        /**
+        * @inheritDoc
+        */
+        public function getDefaultFilters(): ?array
+        {
+            // Make sure newlines are converted to <br /> tags by default
+            return ['nl2br'];
+        }
+        /**
+        * Calculate the total amount of money spent by all customers for all time
+        *
+        * @return float
+        */
+        private function calculateLifetimeSpend(): float
+        {
+            // Add code here to calculate the lifetime spend
+            return 123.45;
+        }
+    }
+   ```
+
+   and save the file to <Vendor>/<module>/Model.
+
+1. Add the new directive to the pool by adding this block to `di.xml`.
+
+   ```xml
+   <config xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="urn:magento:framework:ObjectManager/etc/config.xsd">
+    <type name="Magento\Framework\Filter\SimpleDirective\ProcessorPool">
+        <arguments>
+            <argument name="processors" xsi:type="array">
+                <item name="lifetime_spend" xsi:type="object">GadgetCorp\CustomEmailDirective\Model\LifetimeSpendDirective</item>
+            </argument>
+        </arguments>
+    </type>
+   </config>
+   ```
+
+The new variable is now available within the email template as `{{lifetime_spend}}`.
+Note in the class above, we also defined the filter `shouldBold`. We can use that with `{{lifetime_spend should_bold=1}}`.
+You may also use multiple filters within a var statement: `{{lifetime_spend should_bold=1 |escape}}`.
+
+## Data objects and getUrl
+
+There are a couple of exceptions to strict mode.
+
+One exception is for objects that extend from `DataObject`. These can still be called directly.
+Even then, we do not actually call the getter method directly, but rather, resolve which key is needed and call `getData(<keyname>)` instead.
+
+For example, if we have:
+
+```php
+$template->setVariables(['customer_data'=>new DataObject('mykey' => 'foo')]);
```

Comment:
the key should be my_key not mykey because the call below getMyKey will resolve to getData('my_key') due to the casing.